openmdao/code_review/test_lint_docstrings.py

```diff
@@ -495,95 +504,200 @@ def check_function(self, dir_name, file_name, func_name, func, failures):
+                                   if (inspect.ismethod(getattr(clss, x)) or
+                                       inspect.isfunction(getattr(clss, x))) and
                                    x in clss.__dict__]
+
+                        # Loop over class methods.
                         for method_name in methods:
-                            if print_info:
-                                print(' '*8, method_name)
-                            method = getattr(clss, method_name)
 
-                            self.check_method(dir_name, file_name, class_name,
-                                              method_name, method, failures)
+                            if not method_name.startswith('_'):
+                                full_method_path = f'{module_name}.{class_name}.{method_name}'
+                                try:
+                                    result = validate.validate(full_method_path)
+                                except:
+                                    continue
+
+                                for error_tuple in result['errors']:
+                                    if error_tuple[0] not in ignore:
+                                        if full_method_path not in failures:
+                                            failures[full_method_path] = []
+                                        msg = f"{error_tuple[0]}: {error_tuple[1]}"
+                                        failures[full_method_path].append(msg)
 
-                    # Loop over functions
                     tree = ast.parse(inspect.getsource(mod))
 
                     if hasattr(tree, 'body'):
                         funcs = [node.name for node in tree.body if isinstance(node, ast.FunctionDef)]
                     else:
                         funcs = []
 
+                    # Loop over standalone functions.
                     for func_name in funcs:
                         if not func_name.startswith('_'):
-                            func = getattr(mod, func_name)
-                            self.check_function(dir_name, file_name, func_name,
-                                                func, failures)
+                            full_function_path = f'{module_name}.{func_name}'
+                            try:
+                                result = validate.validate(full_function_path)
+                            except:
+                                continue
+
+                            for error_tuple in result['errors']:
+                                if error_tuple[0] not in ignore:
+                                    if full_function_path not in failures:
+                                        failures[full_function_path] = []
+                                    msg = f"{error_tuple[0]}: {error_tuple[1]}"
+                                    failures[full_function_path].append(msg)
```

Comment:
I think you could factor out this code as a function (used 3 times), passing `full_path` as an arg?

Reply:
Added